packages/backend/src/routers/oauthRouter.ts

```diff
@@ -0,0 +1,302 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import { OAuthIntrospectResponse, oauthPageStyles } from '@lightdash/common';
+import OAuth2Server from '@node-oauth/oauth2-server';
+import express from 'express';
+import { DEFAULT_OAUTH_CLIENT_ID } from '../models/OAuth2Model';
+import { OAuthService } from '../services/OAuthService/OAuthService';
+
+const oauthRouter = express.Router({ mergeParams: true });
+
+// Get OAuth service from request
+function getOAuthService(req: express.Request): OAuthService {
+    return req.services.getOauthService();
+}
+
+// Get authorization - use OAuth2Server
+oauthRouter.get('/authorize', async (req, res, next) => {
+    if (!req.user) {
+        const loginUrl = `/login?redirect=${encodeURIComponent(
+            req.originalUrl || req.url,
+        )}`;
+        return res.redirect(loginUrl);
+    }
+    const {
+        client_id,
+        redirect_uri,
+        scope,
+        state,
+        code_challenge,
+        code_challenge_method,
+    } = req.query;
+    if (!client_id || !redirect_uri || !scope) {
+        return res.status(400).send('Missing required parameters');
+    }
+    // Render authorize page
+    res.set('Content-Type', 'text/html');
```

Comment:
this is missing a lot of the security headers we apply when we server static content from the UI (see the express root router which has the security headers)